### DISCUSSION TITLE
Allow multiple inputs for flow operations

### DISCUSSION THREAD

**Discussion by lukas-schaetzle** [directus/directus#18791] [Feature Requests]
### Summary

Currently, in the flow editor, the input of an operation block only allows to be connected to a single predecessor operation. It would be nice if we could connect multiple blocks to the same block, so we don't have to duplicate the operation for all branches.

### Motivation

In my case, I have a flow where I first try to read the item from the database after the flow has been triggered after updating or creating the same item. I do this to have all fields populated later in the flow and not just the updated ones. However, if the item is created there's no item in the database yet so the operation fails and the flow branches off to the error output from the "read database" operation. If i want the same operations to be executed no matter the result of the previous block, I now have to duplicate all following blocks or start another shared flow. It would be nicer if one could just connect both branches to the same operation.

### Detailed Design

see above

### Must Have:
- Allow multiple flow operations to connect to the same operation 

### Could Have:
- Detect and prevent the creation of loops in the flow editor. Alternatively we could cancel a flow execution after a given timeout

### Drawbacks

This feature would allow users to create endless flow loops.

### Alternatives

Currently you have to duplicate the operations for every branch or you have to start a shared flow in every branch.

### Adoption Strategy

No breaking change

### Unresolved Questions

Prevention of endless loop creation

**Comment by lukas-schaetzle** [comment]
Just noticed there is already a closed discussion for this: [#16356](https://github.com/directus/directus/discussions/16356) (However, the author suggested "A failure and success path can't be merged into one one again". I would also allow it to merge these paths)
And this one is also similar but not exactly the same since it's about multiple outputs instead of inputs: [#17045](https://github.com/directus/directus/discussions/17045)

**Comment by emahuni** [comment]
I came here to ask for the same thing. It makes flows very difficult to write and understand. The only way I had to do it was break down related code into operations that I could then refer to from flows. This is somehow very counter productive.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
